repo_resource/common.py

```diff
@@ -462,15 +464,20 @@ def update_manifest(self, jobs):
 
                 with Pool(jobs) as pool:
                     revisionList = pool.map(multi_run_wrapper, projects)
-                # Convert (remote/project, revision) tuple list
-                # to hash table dict[remote/project]=revision
-                revisionTable = dict((proj, rev) for proj, rev in revisionList)
 
                 # Update revisions
                 for p in manifest.findall('project'):
                     project = p.get('name')
+                    projectBranch = p.get('revision') or defaultBranch
                     projectRemote = p.get('remote') or defaultRemote
-                    p.set('revision', revisionTable[projectRemote+'/'+project])
+                    for url, branch, rev in revisionList:
+                        if (
+                            projectRemote + "/" + project == url
+                            and branch == projectBranch
+                        ):
+                            projectRev = rev
```

Comment:
This part definitely deserves a comment. It might seem obvious but we will forget about this.

Reply:
ok